### ISSUE TITLE
Connection reset after 1.0.4 -> 1.0.5 upgrade

### ISSUE THREAD

**Issue by reverie** [FormidableLabs/urql#242]
Everything's working on 1.0.4, but when I upgrade to 1.0.5, my urql client's graphql query does not complete. On the browser's dev tools, the request Status says '(canceled)' in place of the HTTP code. My backend has Node proxying to Django, and Node's connection to Django errors out with ECONNRESET. Haven't tracked down what's going on in Django yet, but it seems to think it's returning 200. I've confirmed multiple times that it's specific to this urql upgrade.

**Comment by kitten** [comment]
Hiya 👋 that kind of sounds like the query or variables change during your request? On such a case we'd send a teardown and cancel the fetch request.

Do you have some code snippets or a sandbox I could look at?

**Comment by shivekkhurana** [comment]
Hi, I'm facing an issue that might be related to this (using v1.0.5)

I expect the request to teardown since some variables change and the user is redirected, but my problem is that the request is not restarted the next time, 

`fetching` goes from `false` to `true` and stays like that forever. No `error`s, no `data`.

```js
// Query in Parent Component
const [{data, fetching, error}] = useQuery({
    query: fetchOrganisationDocsQuery, variables: {orgId: selectedOrg.id},
    requestPolicy: 'network-only'
  });
```

```js
// onSuccess in a Child Component
const [emptyDocRes, insertEmptyDoc] = useMutation(insertEmptyDocMutation);

  const newDocId = accessByKey(emptyDocRes, 'data.insert_documents.returning.0.id');
  if (newDocId) return (<Redirect push={true} to={`/document/${newDocId}`} />);
```
The parent request is torn down when child redirects. 

But when I do `history.goBack()`, the Parent Component query doesn't run. 

----

Let me know if my issue is related to this thread's issue. If not, I'll open a fresh request.
And many thanks for working on urql. I really like the ability to bisect and use internals. 

🙏

**Comment by reverie** [comment]
Hi, thanks for your help @kitten. Which variables would urql be "watching" for changes? Do you mean the ones passed to the query?

**Comment by kitten** [comment]
So, looking into this, there might be a couple of cases where we're not diffing the input to `useQuery` correctly but that'll be 100% revamped and fixed in v1.1;

It's kind of hard to find out whether this is actually the issue as, assuming the component containing the query unmounts and remounts, queries should always be started on mount.

**Comment by mxstbr** [comment]
I am also running into this. Here is what I see in my DevTools:

![Screenshot 2019-06-05 at 12 01 30](https://user-images.githubusercontent.com/7525670/58948151-c1035080-8789-11e9-8dc5-82c8c8af5b8c.png)

![Screenshot 2019-06-05 at 12 01 40](https://user-images.githubusercontent.com/7525670/58948157-c2347d80-8789-11e9-86a6-f82d130e82d3.png)

My query does _not_ get any variables, so that can't be it.

**Comment by kitten** [comment]
@mxstbr I could release a prerelease with the upcoming fixes so we can check whether this is still an issue potentially?

**Comment by kitten** [comment]
Alright, so chatting with @mxstbr, this is fixed on `master` with commit 950ef7d. There's a prerelease under the `alpha` tag with version `1.1.0-0.alpha` on npm.
